core/src/main/java/com/digitalpebble/stormcrawler/parse/TextExtractor.java

```diff
@@ -138,14 +143,71 @@ public void tail(Node node, int depth) {
                         }
                     }
                 },
-                node);
-        return accum.toString().trim();
+                node,
+                maxTextSize,
+                accum);
     }
 
-    private static void appendNormalisedText(StringBuilder accum, TextNode textNode) {
-        String text = textNode.getWholeText();
+    /**
+     * Start a depth-first traverse of the root and all of its descendants.
+     *
+     * @param visitor Node visitor.
+     * @param root the root node point to traverse.
+     */
+    public static void traverse(
+            NodeVisitor visitor, Node root, int maxSize, StringBuilder builder) {
+        Validate.notNull(visitor);
```

Comment:
We could use `Validate.notNull(Object o, String msg)` to add some context to the `ValidationException` ?